lib/VCMIDirs.cpp

```diff
@@ -170,6 +170,9 @@ class VCMIDirsWIN32 final : public IVCMIDirs
 
 void VCMIDirsWIN32::init()
 {
+	std::locale::global(boost::locale::generator().generate("en_US.UTF-8"));
+	boost::filesystem::path::imbue(std::locale());
```

Comment:
Seems like this line was called for all platforms, except windows. Now for win32 only.

Reply:
This was incorrectly placed ifdef.
If you look up history you'll see that initially this code was added for Windows, and later - first line was gradually removed from other platforms. However it was removed incorrectly - only first line was placed into ifdef

This code is necessary on Windows in order to access files with non-ascii symbols in path (e.g. cyrillic, chinese) using utf-8 since Windows API only allows such access through its utf-16 API (aka wide/unicode API) .
Windows behavior is unique - all other OS's use utf-8 and don't require any special code to allow such access.

Reply:
Does it have something to do with it https://github.com/vcmi/vcmi/pull/873 ?